addons/mrp/models/mrp_production.py

```diff
@@ -438,6 +438,8 @@ def _update_raw_move(self, bom_line, line_data):
         self.ensure_one()
         move = self.move_raw_ids.filtered(lambda x: x.bom_line_id.id == bom_line.id and x.state not in ('done', 'cancel'))
         if move:
+            if 'unit_factor' in line_data:
+                move[0].write({'unit_factor': line_data['unit_factor']})
```

Comment:
remove if condition and add
move[0].write({'unit_factor': quantity/line_data['original_qty']})

Reply:
No, what you do here is incorrect.

"We copy the way the unit_factor is initially computed when updating the lines."
What this means is that if you create a MO for quantity k1, with a unit factor u1, then if you create a MO for quantity k2 then update its quantity to be k1, then the unit factor u2 should be equal to u1.

In the example flow I'm using (which is the one described by the client), I'm getting a unit_factor of 10 with your solution, instead of the correct 0.01.

Reply:
Okay, your patch is OK for me. You can also do a one line patch in _update_raw_move
with move.unit_factor = quantity / move.raw_material_production_id.product_qty
I don't have any preference and I let you choose